rules/no-useless-promise-resolve-reject.js

```diff
@@ -141,14 +157,15 @@ const create = context => {
 
 	return {
 		[selector](callExpression) {
-			const {functionNode, isInTryStatement} = getFunctionNode(callExpression);
-			if (!functionNode || !functionNode.async) {
+			const isReject = callExpression.callee.property.name === 'reject';
+			const {functionNode, isInTryStatement, isInPromiseThenOrCatchCallback} = getFunctionNode(callExpression);
+			if (!functionNode || (!functionNode.async && (!isInPromiseThenOrCatchCallback || isReject))) {
```

Comment:
I don't understand this part, why we should care if it's reject here?

Reply:
My bad, for some reason I thought `promise.catch(() => Promise.reject(error))` would be behave differently to `promise.catch(() => { throw error })`.